### ISSUE TITLE
poetry does not trigger versioning when installed with install-poetry.py

### ISSUE THREAD

**Issue by philippeboyd** [mtkennerly/poetry-dynamic-versioning#61]
Probably linked to issue #34

Steps to reproduce:

```bash
docker run -it --rm python:3.8-slim bash

```
Then inside container:
```bash
apt update && apt install -y curl git
curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python -
export PATH="${HOME}/.local/bin:$PATH"

pip install poetry-dynamic-versioning

cd ~

git clone https://github.com/pypa/sampleproject.git

cd sampleproject

cat << EOF > pyproject.toml
[tool.poetry]
name = "sample"
version = "0.0.0" # used simply as a placeholder for poetry; not the real version
description = "sample project"
authors = ["me"]

[tool.poetry-dynamic-versioning]
enable = true

[tool.poetry.dependencies]
python = ">=3.8,<3.10"

[build-system]
requires = ["poetry-core>=1.0.0", "poetry-dynamic-versioning"]
build-backend = "poetry.core.masonry.api"
EOF

poetry update

poetry build --format wheel --no-interaction
```

Outputs:

```bash
Building sample (0.0.0)
  - Building wheel
  - Built sample-0.0.0-py3-none-any.whl
```

**Instead of something like `0.0.0.postX.dev0+cef9c8c`**

If I run `poetry-dynamic-versioning` and then do the build, it works but changes the `pyproject.toml`; if I have the `dirty` flag, I will have dirty suffix which I don't want.

**Comment by pbarker** [comment]
also seeing this 

> What OS do you use? 

MacOS 11.6

> What Python version do you use? 

3.8.12

> Does pip list mention poetry_dynamic_versioning? 
If so, what version is it? 

0.13.1

> Does poetry run pip list mention poetry_dynamic_versioning?
If so, what version is it? 

it isn't in that list

> What output do you get for these (or the equivalent in your shell):
which python

/Users/pbarker/.pyenv/shims/python

> which pip

/Users/pbarker/.pyenv/shims/pip

> which poetry_dynamic_versioning
not found, but `which poetry-dynamic-versioning` returns `/Users/pbarker/.pyenv/shims/poetry-dynamic-versioning

> which poetry

/Users/pbarker/.local/bin/poetry

> poetry run which python

/Users/pbarker/code/ml-kubeflow-components/.venv/bin/python

> poetry run which pip

/Users/pbarker/code/ml-kubeflow-components/.venv/bin/pip

> poetry run which poetry_dynamic_versioning

/Users/pbarker/.pyenv/shims/poetry-dynamic-versioning

> When you run poetry build, what is the produced wheel file name, and what version do you have in pyroject.toml's [tool.poetry] section?

produces `0.0.0`, the version in the pyproject.toml is also `0.0.0`

**Comment by origliante** [comment]
having the same issue

**Comment by zmarffy** [comment]
The problem seems to lie in installing it from the canonical source, using `curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python -`. Do a `pip install poetry` and it works fine.

**Comment by pbarker** [comment]
> The problem seems to lie in installing it from the canonical source, using `curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python -`. Do a `pip install poetry` and it works fine.

This was true for me as well, thanks @zmarffy

**Comment by zmarffy** [comment]
@philippeboyd Honestly though this isn't a solution to the real problem, you know? We are just treating the symptom, so to speak. I think this issue should still be open as exactly what you described is exactly how you are supposed to install `poetry` on something like a Docker container. It's the "canonical" way.

**Comment by mtkennerly** [comment]
Sorry for the late reply! I did some testing, and it looks like this is a difference between the old get-poetry.py and the new install-poetry.py.

* get-poetry.py would download the Poetry source code, add it to the PYTHONPATH, and run it through the global Python interpreter. This interpreter can see the plugin from `pip install poetry-dynamic-versioning`.
* install-poetry.py creates a virtual environment and runs the interpreter from it. This interpreter cannot see the package installed by `pip install poetry-dynamic-versioning`.

To make it work, you can install the plugin into Poetry's virtual environment:

* Windows: `%APPDATA%\pypoetry\venv\Scripts\pip install poetry-dynamic-versioning`
* Linux: `${XDG_DATA_HOME:-~/.local/share}/pypoetry/venv/bin/pip install poetry-dynamic-versioning`
* Mac (unsure, just inferring from documentation): `~/Library/Application Support/pypoetry/venv/bin/pip install poetry-dynamic-versioning`

After doing that, then `poetry build` will work. Modifying Poetry's virtual environment isn't ideal, but I think it's good enough for now until the official plugin support is released in Poetry 1.2. I'll add this information to the documentation.
